Src/Plugins/Security/SEC101_054.AlibabaAccessKeyValidator.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+// From the Nuget package
+using Aliyun.Acs.Core;
+using Aliyun.Acs.Core.Exceptions;
+using Aliyun.Acs.Core.Profile;
+
+// From the DLL
+using Aliyun.Acs.Iot.Model.V20170420;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class AlibabaAccessKeyValidator : ValidatorBase
+    {
+        internal static AlibabaAccessKeyValidator Instance;
+
+        private const string _keyNotFound = "InvalidAccessKeyId.NotFound";
+        private const string _invalidSecret = "SDK.InvalidAccessKeySecret";
```

Comment:
>_invalidSecret [](start = 29, length = 14)

What is it? An InvalidAccessKey or an InvalidSecret? I.e., you have more terms than you need in this name.

Reply:
These two constant strings themselves are defined in the server and I'm just trying to use them in a switch.  So, you're right, but nothing I can do about it.

---
In reply to: [591881741](https://github.com/microsoft/sarif-pattern-matcher/pull/285#discussion_r591881741) [](ancestors = 591881741)